src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -403,24 +400,38 @@ void GcsServer::StoreGcsServerAddressInRedis() {
   RAY_LOG(INFO) << "Finished setting gcs server address: " << address;
 }
 
-void GcsServer::InitResourceReportPolling(const GcsInitData &gcs_init_data) {
-  gcs_resource_report_poller_.reset(new GcsResourceReportPoller(
-      raylet_client_pool_, [this](const rpc::ResourcesData &report) {
-        gcs_resource_manager_->UpdateFromResourceReport(report);
-      }));
+void GcsServer::InitRaySyncer(const GcsInitData &gcs_init_data) {
+  /*
+    The current synchronization flow is:
+        syncer::poller -> gcs_resource_manager -> syncer::buffer -> syncer::broadcast
+
+    Ideally, we should make it:
+        syncer::poller --> gcs_resource_manager
+                       |-> syncer::buffer -> syncer::broadcast
+    But right now, placement group resource update is preventing us from doing this.
+    TODO (iycheng): Update placement group resource reporting and update it to the
+    right flow.
+  */
+  auto on_update = [this](const rpc::ResourcesData &report) {
+    main_service_.dispatch(
```

Comment:
what's the difference between dispatch vs post?

Reply:
if it's already running in the event loop, it'll just execute this instead of posting to the queue.